VideoLocker/src/main/java/org/edx/mobile/util/ResourceUtil.java

```diff
@@ -1,64 +1,59 @@
 package org.edx.mobile.util;
 
 import android.content.Context;
+import android.support.annotation.PluralsRes;
+import android.support.annotation.StringRes;
 
 import com.squareup.phrase.Phrase;
 
-import org.edx.mobile.R;
 import org.edx.mobile.base.MainApplication;
 
+import java.util.HashMap;
+import java.util.Set;
+
 
 public class ResourceUtil {
     public static final String QuantityHolder = "quantity";
 
-
-    public static String getResourceString(String name) {
-        Context context = MainApplication.instance().getApplicationContext();
-        return getResourceString(name, context);
-    }
-
-    /**
-     * get the string resources dynamically
-     * @param name
-     * @param context
-     * @return
-     */
-    public static String getResourceString(String name, Context context) {
-        int nameResourceID = context.getResources().getIdentifier(name, "string", context.getApplicationInfo().packageName);
-        if (nameResourceID == 0) {
-            throw new IllegalArgumentException("No resource string found with name " + name);
-        } else {
-            return context.getString(nameResourceID);
-        }
-    }
-
-    public static String getResourceString(int resourceId){
+    public static String getResourceString(@StringRes int resourceId) {
         Context context = MainApplication.instance().getApplicationContext();
         return context.getString(resourceId);
     }
 
-    public static CharSequence getFormattedString(int resourceId, String key, String value){
-        if ( value == null )
+    public static CharSequence getFormattedString(@StringRes int resourceId, String key, String value) {
+        if (value == null)
             value = "";
         return Phrase.from(ResourceUtil.getResourceString(resourceId))
-                .put(key, value) .format();
+                .put(key, value).format();
     }
 
+    public static CharSequence getFormattedString(@StringRes int resourceId, HashMap<String, CharSequence> keyValMap) {
```

Comment:
Instead of requiring a `HashMap` for formatting strings with multiple keys, we might consider using a string varargs parameter instead, which would be cheaper to initialize and result in neater code. It would also require only one method for single or multiple key-value pairs:

````java
public static CharSequence getFormattedString(Context context,
        @StringRes int resourceId, String key, String value, String... keyValuePairs) {
    Phrase phrase = Phrase.from(context, resourceId);
    if (key == null) {
        throw new NullPointerException("Key can't be null");
    }
    phrase.put(key, value == null ? "" : value);
    if (keyValuePairs != null) {
        if (keyValuePairs.length % 2 != 0) {
            throw new IllegalArgumentException(
                    "The key value pairs must be an even number of items.");
        }
        for (int i = 0; i < keyValuePairs.length - 1;) {
            key = keyValuePairs[i++];
            if (key == null) {
                throw new NullPointerException("Key can't be null");
            }
            value = keyValuePairs[i++];
            phrase.put(key, value == null ? "" : value);
        }
    }
    return phrase.format();
}
````